File: networklogger/src/main/java/com/network/logger/detail/NetworkLoggerDetailActivity.java
package com.network.logger.detail;


import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.network.logger.R;
import com.network.logger.database.AppDatabase;
import com.network.logger.util.Constant;

public class NetworkLoggerDetailActivity extends AppCompatActivity implements NetworkLoggerDetailView {

    private NetworkLoggerDetailPresenter presenter;

    private TextView tvData;
    private ImageView ivBack;
    private ImageView ivShare;
    private ProgressBar progressBar;

    private String mData = "";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_network_logger_detail);

        int uid = getIntent().getIntExtra(Constant.UID, 0);

        setView();
        setAction();

        presenter = new NetworkLoggerDetailPresenter(this, this, AppDatabase.getAppDatabase());
        presenter.getData(uid);
    }

    private void setView() {
        tvData = findViewById(R.id.tvData);
        ivBack = findViewById(R.id.ivBack);
        ivShare = findViewById(R.id.ivShare);
        progressBar = findViewById(R.id.progressBar);
    }

    private void setAction() {
        ivBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });

        ivShare.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ivShare.setVisibility(View.GONE);
                progressBar.setVisibility(View.VISIBLE);
                Intent sendIntent = new Intent();
                sendIntent.setAction(Intent.ACTION_SEND);
                sendIntent.putExtra(Intent.EXTRA_TEXT, mData);
                sendIntent.setType("text/plain");
                Intent shareIntent = Intent.createChooser(sendIntent, null);
                startActivity(shareIntent);
            }
        });
    }

    @Override
    public void showData(String data) {
        mData = data;
        tvData.setText(mData);
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (ivShare != null && progressBar != null){
            ivShare.setVisibility(View.VISIBLE);
            progressBar.setVisibility(View.GONE);
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        presenter.unBind();
    }
}

File: networklogger/src/main/java/com/network/logger/detail/NetworkLoggerDetailActivity.kt
package com.network.logger.detail

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.network.logger.database.AppDatabase
import com.network.logger.databinding.ActivityNetworkLoggerDetailBinding
import com.network.logger.util.Constant
import kotlinx.coroutines.launch


class NetworkLoggerDetailActivity : AppCompatActivity(), NetworkLoggerDetailView {

    private var binding: ActivityNetworkLoggerDetailBinding? = null
    private var presenter: NetworkLoggerDetailPresenter? = null
    private var mData = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityNetworkLoggerDetailBinding.inflate(layoutInflater)
        setContentView(binding?.root)

        val uid = intent.getIntExtra(Constant.UID, 0)

        setAction()

        presenter = NetworkLoggerDetailPresenter(this, AppDatabase.getAppDatabase())
        lifecycleScope.launch { presenter?.getData(uid) }
    }

    private fun setAction() {
        binding?.ivBack?.setOnClickListener { finish() }

        binding?.ivShare?.setOnClickListener {
            binding?.ivShare?.visibility = View.GONE
            binding?.progressBarShare?.visibility = View.VISIBLE
            val sendIntent = Intent()
            sendIntent.setAction(Intent.ACTION_SEND)
            sendIntent.putExtra(Intent.EXTRA_TEXT, mData)
            sendIntent.setType("text/plain")
            val shareIntent = Intent.createChooser(sendIntent, null)
            startActivity(shareIntent)
        }
    }

    override fun showData(data: String?) {
        mData = data.toString()
        binding?.progressBar?.visibility = View.GONE
        binding?.tvData?.text = mData
    }

    override fun onResume() {
        super.onResume()
        binding?.ivShare?.visibility = View.VISIBLE
        binding?.progressBarShare?.visibility = View.GONE
    }

    override fun onDestroy() {
        super.onDestroy()
        presenter?.unBind()
    }
}
